### PR TITLE
pull/checkout: fix empty log output if no state gets changed

### PR DESCRIPTION
**Pull request by skshetry**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here. If the CLI API is changed, I have updated [tab completion scripts](https://github.com/iterative/dvc/tree/master/scripts/completion).

* [x] ❌ I will check DeepSource, CodeClimate, and other sanity checks below. (We consider them recommendatory and don't expect everything to be addressed. Please fix things that actually improve code or fix bugs.)

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/command/checkout.py [hunk 1]
```diff
@@ -26,12 +26,13 @@ def _human_join(words):
...
-    summary = (
+    summary = [
         "{} {}".format(len(stats[state]), state)
         for state in states
         if stats.get(state)
-    )
-    logger.info(_human_join(summary))
+    ]
+    if summary:
+        logger.info(_human_join(summary))
```
Comment (efiop) [inline dvc/command/checkout.py]: Maybe it is worth putting some message that will say that there was nothing to do?

Comment (skshetry) [reply dvc/command/checkout.py]: Makes sense. Thanks.

File: tests/unit/command/test_checkout.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def _assert_output(stats, expected_text):
     del stats["modified"]
     _assert_output(stats, "3 added")
 
-    _assert_output({}, "")
```
Comment (skshetry) [inline tests/unit/command/test_checkout.py]: This was a :snake::oil_drum: assertion. :slightly_frowning_face:

**Comment by skshetry** [comment]
We should fix `pull` in this next sprint. The output will be very confusing since `fetch` won't print a thing if fetched something.

#### When nothing was fetched and not checked out:
```
$ dvc pull
No changes.
Everything is up to date.
```

#### When something was fetched, but not checked out.
```
$ dvc pull
No changes.
```

#### When something was fetched + checked out:
```
$ dvc pull
3 added
```

#### When something was not fetched, but checkout was done:
```
$ dvc pull
3 added
Everything is up to date.
```

**Comment by efiop** [comment]
@skshetry Please create a ticket and add it to the next sprint.

**Comment by efiop** [comment]
@skshetry Also, what about docs?

**Pull request merged**